Fixing 693 & 687

charbel14 opened the pull request:
Due to a change in jqueryui 1.11.2 going forward the _find no longer returns the tip only but the element as well.

Changed the tip function to return the tip is there or else an empty jquery element.

charbel14 committed:
Update jqueryui-editable.js

Trying a fix for issue 693.

charbel14 committed:
Update jqueryui-editable.js

Testing another fix

charbel14 committed:
Update jqueryui-editable.js

charbel14 committed:
Update jqueryui-editable.js

charbel14 committed:
Update jqueryui-editable.js

charbel14 committed:
Update jqueryui-editable.js

charbel14 committed:
Fixing issue 693 and 687.

charbel14 committed:
Fixing issue 693 & 687

charbel14 commented:
The minified version will need to be regenerated.

hacfi commented on dist/jqueryui-editable/js/jqueryui-editable.js:
```diff
@@ -4718,7 +4718,12 @@ Editableform based on jQuery UI
         
         tip: function() {
-            return this.container() ? this.container()._find(this.container().element) : null;
+            var container = this.container();
```
There is one accidental space before `var`.

hacfi commented on src/containers/editable-tooltip.js:
```diff
@@ -57,7 +57,12 @@
-            return this.container() ? this.container()._find(this.container().element) : null;
+            var container = this.container();
+            if (container) {
+            var tip = container._find(this.container().element);
```
This line should be indented.

hacfi commented:
I think there is an issue with spaces/tabs. Besides that it looks good to me!

charbel14 committed:
Update jqueryui-editable.js

charbel14 committed:
Adding support to allow for a promise as source

Source can now be a promise.

charbel14 committed:
Updating the source response to accept a promise.

horizons2 commented:
tested with jquery.ui v1.11.2 works again, please merge.

NotionCommotion commented:
tested with jquery.ui v1.11.2 works.  Tested with jquery.ui v1.11.4, and doesn't work.

charbel14 committed:
Select2 is destroyed when you click on input field

Added a select2 entry into the execlude class list. This is so  that the editor would not get destroyed when mouse click in select2 input search element

charbel14 committed:
Added a subversion to the patch. 

Hopefully will help in getting latest when you run bower install.

charbel14 committed:
buming the subversion 

this is for patch-1

charbel14 committed:
bumping version to 1.5.1.1

tofutim commented:
Can you redo the PR into 'dev'?

tofutim commented:
any update on 11.4 compatibility? also see issue #826

charbel14 committed:
Correcting the version to 1.5.1-1

MarQuisKnox reviewed (changes requested):
1. Please remove the version number changes
2. Please submit this PR to the `develop` branch

Thanks.

kakoma commented:
Used `1.11.0`. Anything higher failed in my tests

mlisowsk commented:
I confirm that applying commit https://github.com/vitalets/x-editable/pull/724/commits/4ef26e2e91ae5060a7588aea55b330b51b04598d to the source and then rebuilding with `grund build` fixes the issue with popup text not working in modern jQuery-UI due to an exception related to the tip() method returning `undefined`.

Tested and works with jQuery-UI 1.13.2 and jQuery 3.7.1  (tested only the text INPUT as popup and inline).
